crates/xtask/src/changelog.rs

```diff
@@ -0,0 +1,377 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+use anyhow::{ensure, Context, Result};
+use clap::ValueEnum;
+
+#[derive(clap::ValueEnum, Clone, Debug)]
+pub enum ChangelogReleaseType {
+    Major,
+    Minor,
+    Patch,
+}
+
+/// Changelog file data structure.
+#[derive(Debug, Default, PartialEq)]
+struct Changelog {
+    releases: Vec<ChangelogRelease>,
+
+    increment_count: u32,
+}
+
+impl Changelog {
+    fn push_release(&mut self, release: ChangelogRelease) {
+        self.releases.push(release);
+    }
+
+    fn set_increment_count(&mut self, num: u32) {
+        self.increment_count = num;
+    }
+}
+
+/// Each release of a Changelog, containing a version string and optional major, minor, patch change
+/// descriptions.
```

Comment:
Let's try to keep the title short. But I would also not be against having no doc. This is self-describing. (Same for the `Changelog` type actually).